src/blueprint/azext_blueprint/_params.py

```diff
@@ -0,0 +1,215 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+from azure.cli.core.commands.parameters import (
+    tags_type,
+    get_enum_type,
+    get_location_type
+)
+from knack.arguments import CLIArgumentType
+from argcomplete.completers import FilesCompleter
+from ._validators import scope_validator
+
+parameter_type = CLIArgumentType(
+    options_list=['--parameters'],
+    help='Parameters in JSON format or file path to Json file with "@" prefix.'
+)
+
+template_type = CLIArgumentType(
+    options_list=['--template'],
+    help='ARM template in JSON format or file path to Json file with "@" prefix.'
+)
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('blueprint', validator=scope_validator) as c:
+        c.argument('management_group', help='The management group for the scope of the resource. The scope could either be a subscription or a management group. For blueprint assignments management group scope is reserved for future use. If management-group is not specified, will use subscription as scope')

```

Comment:
For management_group parameter, should it be manage group name or ID? And the help message is a little confusing with "The scope could be a subscription or a management group...".

Reply:
Not sure if it's the same management group, but the management group id in blueprint is not supported in parse_resource_id. I've submitted an issue for msrestazure: https://github.com/Azure/msrestazure-for-python/issues/139

For scope, blueprint swagger defines a `scope` parameter, it could either be a management group or a subscription. Powershell splited it into two parameters, and I followed their design since we can make use of the global argument --subscription. The logic to handle these 2 parameters is that if --management-group is provided, we ignore the --subscription, juse use use management-group for scope value, otherwise we fill the scope value from --subscription or the default subscription if both parameters are not provided.

Reply:
For management group value, it is something related to tanant, not subscription. CLI authentication is based on default subscription information. I know our target here is bluprint client. But I am thinking if we need to confirm we have the access permission for management group or the subscription. If we don;t have the permission, should the specified scope be invalid?

In additon, should we consider grouping `--management-scope` and `--subscription` as `scope` argument group? Ad you know, currently --subscription is global argument part and it is hard for users to associate the --subcription to scope parameters.

I am just trying to come up with some potential issues. If it's fine to your service, feel free to let me know.

Reply:
We can add more sanity checks in depth coverage. Let's keep it simple in breadth coverage. The command could be used as a downstream to commands to first get one' management groups(we do have such commands already, right?), then pass that group here. 

As for scope, I splited it because first Powershell designed it this way and PMs want us to be consistent with Powershell, second if it's scope, you need to pass the full resource id of subscription or management group, with msrestazure not able to parse management group id now, it may not be convenient for users.